src/device-state/current-state.ts

```diff
@@ -86,13 +93,16 @@ const sendReportPatch = async (
 		body,
 	};
 
-	const [{ statusCode, body: statusMessage }] = await request
-		.patchAsync(endpoint, params)
-		.timeout(apiTimeout);
+	const [
+		{ statusCode, body: statusMessage, headers },
+	] = await request.patchAsync(endpoint, params).timeout(apiTimeout);
 
 	if (statusCode < 200 || statusCode >= 300) {
```

Comment:
The current condition allows 300 for some reason.

```suggestion
// If statusCode is not 2xx range then throw an error
	if (statusCode < 200 || statusCode > 300) {
```

Reply:
300 is an error response, so the code is correct no? https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/300

Reply:
ah, I had a brain fart. Yes we WANT to include 300... this would allow 200...300 but I meant 200..299.

Can you add the comment though, I think it helps